core/src/main/scala/org/bitcoins/core/wallet/builder/RawTxBuilder.scala

```diff
@@ -0,0 +1,263 @@
+package org.bitcoins.core.wallet.builder
+
+import org.bitcoins.core.number.{Int32, UInt32}
+import org.bitcoins.core.protocol.script.{
+  CLTVScriptPubKey,
+  CSVScriptPubKey,
+  EmptyScriptSignature
+}
+import org.bitcoins.core.protocol.transaction.{
+  BaseTransaction,
+  EmptyWitness,
+  NonWitnessTransaction,
+  Transaction,
+  TransactionConstants,
+  TransactionInput,
+  TransactionOutput,
+  WitnessTransaction
+}
+import org.bitcoins.core.wallet.utxo.{
+  ConditionalInputInfo,
+  EmptyInputInfo,
+  InputInfo,
+  InputSigningInfo,
+  LockTimeInputInfo,
+  MultiSignatureInputInfo,
+  P2PKHInputInfo,
+  P2PKInputInfo,
+  P2PKWithTimeoutInputInfo,
+  P2SHInputInfo,
+  P2WPKHV0InputInfo,
+  P2WSHV0InputInfo,
+  UnassignedSegwitNativeInputInfo
+}
+
+import scala.annotation.tailrec
+import scala.collection.mutable
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success, Try}
+
+class RawTxBuilder {
+  private var version: Int32 = TransactionConstants.validLockVersion
+
+  private val inputsBuilder: mutable.Builder[
+    TransactionInput,
+    Vector[TransactionInput]] = Vector.newBuilder
+
+  private val outputsBuilder: mutable.Builder[
+    TransactionOutput,
+    Vector[TransactionOutput]] = Vector.newBuilder
```

Comment:
I think the tx builder should have some distinction from change outputs and outputs we are sending to. This will make things simpler for reasoning things like changing the fee rate of the tx or combining 2 tx builders, etc

Reply:
Will add docs once things are finalized but the outputs here are all non-change so this should work out +1